app/assets/javascripts/oxalis/view/settings/merger_mode_modal_view.js

```diff
@@ -0,0 +1,58 @@
+// @flow
+import * as React from "react";
+import { Modal, Button, Spin, Tooltip } from "antd";
+
+type Props = {
+  isCloseable: boolean,
+  onClose: () => void,
+};
+
+export default function MergerModeModalView({ isCloseable, onClose }: Props) {
+  return (
+    <Modal
+      visible
+      title="Merger mode enabled"
+      closable={false}
+      centered
+      footer={
+        <div className="centered-children">
+          {!isCloseable ? (
+            <Tooltip title="At the moment, the existing trees are used to merge segments. This dialog can be closed after the initial processing has been completed.">
+              <Button type="primary" onClick={onClose} disabled>
+                Close
+              </Button>
+            </Tooltip>
+          ) : (
+            <Button type="primary" onClick={onClose}>
```

Comment:
I tried this, but the Tooltip has to be the parent of the button when it shall be displayed on hover of the button. As I do not want the tooltip to be rendered when the processing is completely done, I only render the Tooltip conditionally.
I now used a function that returns the button. Whether the button is disabled or not is defined by a parameter.

Reply:
Ah, I see, my bad. Your new solution is fine :+1: You could transform the function to a simple expression since the parameter is known beforehand, but as I said the current solution is fine, too.

```
  const closeButton = <Button type="primary" onClick={onClose} disabled={!isCloseable}>
      Close
    </Button>;

  return (
    <Modal
      visible
      title="Merger mode enabled"
      closable={false}
      centered
      footer={
        <div className="centered-children">
          {!isCloseable ? (
            <Tooltip title="At the moment, the existing trees are used to merge segments. This dialog can be closed after the initial processing has been completed.">
              {closeButton}
            </Tooltip>
          ) : (
            closeButton
          )}
        </div>
      }
    >
```